subcommand/lifecycle-sidecar/command_test.go

```diff
@@ -36,10 +36,19 @@ func testRunSignalHandling(sig os.Signal) func(*testing.T) {
 		tmpDir, configFile := createServicesTmpFile(t, servicesRegistration)
 		defer os.RemoveAll(tmpDir)
 
+		a, err := testutil.NewTestServerConfigT(t, nil)
+		require.NoError(t, err)
+		defer a.Stop()
+
 		ui := cli.NewMockUi()
 		cmd := Command{
 			UI: ui,
 		}
+
+		client, err := api.NewClient(&api.Config{
+			Address: a.HTTPAddr,
+		})
+		require.NoError(t, err)
```

Comment:
This test still passes if I revert the changes made because the command isn't given the http-addr so it can't register the services regardless of the signal handling.

We just need to add `-http-addr` and `-sync-period` flags to the line below:
```go
		exitChan := runCommandAsynchronously(&cmd, []string{
			"-service-config", configFile,
			"-http-addr", a.HTTPAddr,
			"-sync-period", "1s",
		})
		cmd.sendSignal(sig)
```

And then the test fails without the changes and passes with them.